anasymod/templates/zynq_gpio.py

```diff
@@ -77,43 +92,110 @@ def __init__(self, ps_vlnv='xilinx.com:ip:processing_system7:5.5',
     [get_bd_ports i_data] \\
     [get_bd_pins axi_gpio_1/gpio2_io_o]
 
-# Apply automation    
+####################
+# Apply automation #
+####################   
+
+# processing system automation
 
+{% if subst.is_ultrascale %}
+apply_bd_automation \\
+    -rule xilinx.com:bd_rule:zynq_ultra_ps_e \\
+    -config { \\
+        apply_board_preset "1" \\
+    } \\
+    [get_bd_cells zynq_ultra_ps_e_0]
+{% else %}
 apply_bd_automation \\
     -rule xilinx.com:bd_rule:processing_system7 \\
-    -config {{ \\
+    -config { \\
         make_external "FIXED_IO, DDR" \\
         apply_board_preset "1" \\
         Master "Disable" \\
         Slave "Disable" \\
-    }} \\
+    } \\
     [get_bd_cells processing_system7_0]
+{% endif %}
 
+# axi_gpio_0 automation

```

Comment:
For now, block diagram automation is basically specially-cased for UltraScale+ vs. other designs.  Maybe there is a way to consolidate the two, since there are just a couple of small differences.